File: server/src/main/java/com/krillsson/sysapi/core/metrics/CpuMetrics.java
package com.krillsson.sysapi.core.metrics;

import com.krillsson.sysapi.core.domain.cpu.CpuInfo;
import com.krillsson.sysapi.core.domain.cpu.CpuLoad;

public interface CpuMetrics {
    CpuInfo cpuInfo();

    CpuLoad cpuLoad();

    long uptime();
}

File: src/main/kotlin/com/krillsson/sysapi/core/metrics/CpuMetrics.kt
package com.krillsson.sysapi.core.metrics

import com.krillsson.sysapi.core.domain.cpu.CpuInfo
import com.krillsson.sysapi.core.domain.cpu.CpuLoad
import reactor.core.publisher.Flux

interface CpuMetrics {
    fun cpuInfo(): CpuInfo
    fun cpuLoad(): CpuLoad
    fun cpuLoadEvents(): Flux<CpuLoad>
    fun uptime(): Long
}